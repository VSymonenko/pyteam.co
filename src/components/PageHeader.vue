<template lang="pug">
  v-flex(:style={ display: 'flex', alignItems: 'center', flexDirection: 'column' })
    div(class='d-flex display-3') {{ header }}
    div(class='line')
    div(class='d-flex display-1' :style={ color: '#777777'} v-if='subHeader') {{ subHeader }}
</template>

<script lang="ts">
import Vue from 'vue';
import Component from 'vue-class-component';
import { NoCache } from '@/decorators';

@Component({
  name: 'PageHeader',
  props: {
    header: {
      type: String,
    },
    subHeader: {
      type: String,
    },
  },
})

export default class PageHeader extends Vue {
}
</script>

<style lang="stylus" scoped>
.line
  background-color #777777
  height 1px
  width 15vh
  margin-bottom 10px
</style>
