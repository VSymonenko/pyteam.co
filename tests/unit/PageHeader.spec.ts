import PageHeader from '@/components/PageHeader.vue';
import { shallowMount } from '@vue/test-utils';
import './common';

test('renders correctly', () => {
  const wrapper = shallowMount(PageHeader);

  expect(wrapper.isVueInstance()).toBeTruthy();
  expect(wrapper).toMatchSnapshot();
});

describe('PageHeader', () => {
  it('renders header in PageHeader', () => {
    const header = 'new header';
    const wrapper = shallowMount(PageHeader, {
      propsData: { header },
    });
    expect(wrapper.props().header).toContain(header);
  });

  it('renders a subHeader in PageHeader', () => {
    const subHeader = 'new sub-header';
    const wrapper = shallowMount(PageHeader, {
      propsData: { subHeader },
    });
    expect(wrapper.props().subHeader).toContain(subHeader);
  });
});
