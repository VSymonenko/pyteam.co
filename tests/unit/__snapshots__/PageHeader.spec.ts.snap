// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<vflex-stub tag="div">
  <div class="d-flex display-3"></div>
  <div class="line"></div>
  <!---->
</vflex-stub>
`;
